test/v3/controllers/requestParsers/validators/RetrieveUkPropertyValidatorSpec.scala

```diff
@@ -32,23 +32,23 @@ class RetrieveUkPropertyValidatorSpec extends UnitSpec {
   "validator" should {
     "return no errors" when {
       "passed valid raw data" in {
-        val input = RetrieveUkPropertyBsasRawData(validNino, validCalculationId)
+        val input = RetrieveUkPropertyBsasRawData(validNino, validCalculationId, taxYear = None)
         validator.validate(input) shouldBe List()
       }
     }
     "return a single error" when {
       "passed raw data with an invalid nino" in {
-        val input = RetrieveUkPropertyBsasRawData(invalidNino, validCalculationId)
+        val input = RetrieveUkPropertyBsasRawData(invalidNino, validCalculationId, taxYear = None)
         validator.validate(input) shouldBe List(NinoFormatError)
       }
       "passed raw data with an invalid bsas id" in {
-        val input = RetrieveUkPropertyBsasRawData(validNino, invalidCalculationId)
+        val input = RetrieveUkPropertyBsasRawData(validNino, invalidCalculationId, taxYear = None)
         validator.validate(input) shouldBe List(CalculationIdFormatError)
       }
     }
     "return multiple errors" when {
       "passed raw data with multiple invalid fields" in {
-        val input = RetrieveUkPropertyBsasRawData(invalidNino, invalidCalculationId)
+        val input = RetrieveUkPropertyBsasRawData(invalidNino, invalidCalculationId, taxYear = None)
```

Comment:
after adding the taxYear format parser. 
You will need to add the following test 
https://github.com/hmrc/self-assessment-bsas-api/blob/main/test/v3/controllers/requestParsers/validators/RetrieveSelfEmploymentValidatorSpec.scala#L59